### PR TITLE
Add Overall difficulty article

### PR DESCRIPTION
**Pull request by Pennek**
Resolves part of #3151

### CODE REVIEW

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+tags:
+- OD
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
  - OD
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- Hit Window
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
  - Hit Window
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+- Notelock
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
  - Notelock
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+# Overall Difficulty
+
+*For recommended OD values, see: [Ranking Criteria](/wiki/Ranking_Criteria/)*
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
*For recommended OD values, see: [Ranking Criteria](/wiki/Ranking_Criteria)*
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+Overall difficulty, commonly abbreviated as *OD*, defines how difficult it is to achieve high accuracy. The value ranges from 0 to 10; higher OD values will require more accuracy and precision. At OD0, a player can get a hit score of 300 at less than 79.5ms away from hitting it exactly on-time. On the other end of the scale, OD10 requires being less than 19.5ms away for the same hit score of 300.
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
Overall difficulty, commonly abbreviated as *OD*, defines how difficult it is to achieve high [accuracy](/wiki/Accuracy). The value ranges from 0 to 10; higher OD values will require more accuracy and precision. At OD0, a player can get a hit score of 300 at less than 79.5ms away from hitting it exactly on-time. On the other end of the scale, OD10 requires being less than 19.5ms away for the same hit score of 300.
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Since accuracy is important for gaining health, overall difficulty indirectly influences how hard it is to pass a [beatmap](/wiki/Beatmaps). Higher OD values mean shorter time windows to clear [hit objects](/wiki/Hit_Objects), both in general and in terms of getting high score values. The width of the hit window is defined by the following table:
+
+| Score | Hit Window |
+| --- | --- |
+| 50 | `150ms + 50ms * (5 - OD) / 5` |
+| 100 | `100ms + 40ms * (5 - OD) / 5` |
+| 300 | `50ms + 30ms * (5 - OD) / 5` |
+
+In osu!standard, overall difficulty also affects spinners, in which they must be spun more to fill up the gauge in time. In osu!taiko, the denden will also need more hits to clear it. The difficulty of spinners is defined through the following formulas:
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
In [osu!standard](/wiki/Game_Modes/osu!), overall difficulty also affects spinners, in which they must be spun more to fill up the gauge in time. In [osu!taiko](/wiki/Game_Modes/osu!taiko), the denden will also need more hits to clear it. The difficulty of spinners is defined through the following formulas:
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+---
+tags:
+- OD
+- Hit Window
+- Notelock
+---
+
+# Overall Difficulty
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]: Also change folder caps. If you can't change folder caps locally, try `git config core.ignorecase false`.
```suggestion
# Overall difficulty
```

File: wiki/Beatmapping/Overall_Difficulty/en.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+- OD = 5: `spins_per_second = 5`
+- OD > 5: `spins_per_second = 5 + 2.5 * (OD - 5) / 5`
+
+## Notelock
+
+In osu!standard, the timing system will not allow a hit object to be tapped until the previous object is tapped (or missed after the hit window passes). This can lead to instances of *notelock* where players will miss a note even though they hit it perfectly in sync with the beat. This can happen if a beatmap uses a low OD value. With low OD values, the hit window of one hit object may overlap the next. Thus, a player could hit the second hit object with perfect timing after failing to hit the first hit object. This would result in completely missing both hit objects due to the hit window of the first hit object being active as the second is to be tapped.
+
+## Mods
+
+There are four mods which alter Overall Difficulty timing when activated:
```
Comment (Joehuu) [inline wiki/Beatmapping/Overall_Difficulty/en.md]:
```suggestion
There are four mods which alter overall difficulty timing when activated:
```

File: wiki/Beatmapping/Overall_Difficulty/en.md (hunk 7 above)
Comment (cl8n) [reply wiki/Beatmapping/Overall_Difficulty/en.md]: on windows i find `git mv` a lot more straightforward, unsetting `core.ignorecase` can potentially make hell of your index if you try to checkout across branches where caps change and your workdir doesnt exactly match what git was expecting

idk maybe not relevant here just warning u lol

**Comment by cl8n** [comment]
> For consistency reasons, what should we do with "Accuracy" web side?

seems fine as Accuracy there since that's most of what it does anyway, and "overall difficulty" doesn't really say anything. here I think you could add accuracy as  a tag.

**Pull request merged**